tests/setupTests.js

```diff
@@ -1,43 +1,52 @@
 /* global jest, expect, beforeEach, afterAll */
 
 const crypto = require(`crypto`);
+const process = require(`process`);
 
 jest.retryTimes(2, {logErrorsBeforeRetry: true});
 
 switch (process.env.NOCK_ENV || ``) {
   case `record`:
   case `replay`:
     beforeEach(() => {
-      // To ensure we test the default behavior, we must remove these env vars
-      // in case the local machine already set these values.
-      delete process.env.COREPACK_DEFAULT_TO_LATEST;
-      delete process.env.COREPACK_ENABLE_NETWORK;
-      delete process.env.COREPACK_ENABLE_PROJECT_SPEC;
-      delete process.env.COREPACK_ENABLE_STRICT;
-      delete process.env.COREPACK_HOME;
-      delete process.env.COREPACK_NPM_REGISTRY;
-      delete process.env.COREPACK_NPM_TOKEN;
-      delete process.env.COREPACK_NPM_USERNAME;
-      delete process.env.FORCE_COLOR;
+      const {
+        // To ensure we test the default behavior, we must remove these env vars
+        // in case the local machine already set these values.
+        COREPACK_DEFAULT_TO_LATEST,
+        COREPACK_ENABLE_NETWORK,
+        COREPACK_ENABLE_PROJECT_SPEC,
+        COREPACK_ENABLE_STRICT,
+        COREPACK_HOME,
+        COREPACK_NPM_REGISTRY,
+        COREPACK_NPM_TOKEN,
+        COREPACK_NPM_USERNAME,
+        FORCE_COLOR,
+        // We save the rest to put it into `testEnv`.
+        ...processEnv
+      } = process.env;
 
-      process.env.RUN_CLI_ID = 0;
-      process.env.NOCK_FILE_NAME = crypto
-        .createHash(`md5`)
-        .update(expect.getState().currentTestName)
-        .digest(`base64url`);
-    });
-
-    afterAll(() => {
-      delete process.env.RUN_CLI_ID;
-      delete process.env.NOCK_FILE_NAME;
+      process.testEnv = {
```

Comment:
Why not continue using `process.env`?

My suggestion is something like:
```js
let env;
beforeEach(() => {
  env = {...process.env};
});

afterEach(() => {
  process.env = env;
});
```

Reply:
> Why not continue using `process.env`?

What I don't like about that is that it feels "magic", someone new to the codebase would likely not be aware of that.

> If a test modifies `process.env` instead of `process.testEnv` we'll have the same problem as before.

That seems rare enough that I would be okay with having a bit of clean up code to handle those use case.

Reply:
But the code we're testing uses `process.env` so we have to use it either way.

Reply:
Not all of the tests do that, for example https://github.com/nodejs/corepack/blob/863e0e0a500a39f244d1ef0c0fa5e15c0787c345/tests/npmRegistryUtils.test.ts

Reply:
Yeah and this one is handling it correctly

Reply:
Yes, and that's how I would prefer the other test files be handled as well.

A setup file would avoid the repetition but if you'd rather have it in each test file then that's fine by me.

Reply:
Alright, I've settled on adding comments in test files.